From 000d48992bc7b9bfc8956f5393dcc83d0126a801 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 29 Jun 2019 10:55:32 +0800
Subject: [PATCH] kickstart: not support Authselect/Firewall/Firstboot

We do not support kickstart for now

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Refactor for version anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 anaconda.py                | 2 +-
 pyanaconda/installation.py | 6 +++---
 pyanaconda/kickstart.py    | 5 +++++
 3 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/anaconda.py b/anaconda.py
index cefe6849f..cbff33eaf 100755
--- a/anaconda.py
+++ b/anaconda.py
@@ -428,7 +428,7 @@ if __name__ == "__main__":
     anaconda.payload.set_from_opts(opts)
 
     # Initialize the security configuration.
-    startup_utils.initialize_security()
+    #startup_utils.initialize_security()
 
     # Set the language before loading an interface, when it may be too late.
     startup_utils.initialize_locale(opts, text_mode=anaconda.tui_mode)
diff --git a/pyanaconda/installation.py b/pyanaconda/installation.py
index 705f1b522..95f67a43a 100644
--- a/pyanaconda/installation.py
+++ b/pyanaconda/installation.py
@@ -93,7 +93,7 @@ def _prepare_configuration(payload, ksdata):
     os_config = TaskQueue("Installed system configuration", N_("Configuring installed system"))
 
     # add installation tasks for the Security DBus module
-    if is_module_available(SECURITY):
+    if False:
         security_proxy = SECURITY.get_proxy()
         security_dbus_tasks = security_proxy.InstallWithTasks()
         os_config.append_dbus_tasks(SECURITY, security_dbus_tasks)
@@ -118,7 +118,7 @@ def _prepare_configuration(payload, ksdata):
         os_config.append_dbus_tasks(LOCALIZATION, localization_dbus_tasks)
 
     # add the Firewall configuration task
-    if conf.target.can_configure_network:
+    if False:
         firewall_proxy = NETWORK.get_proxy(FIREWALL)
         firewall_dbus_task = firewall_proxy.InstallWithTask()
         os_config.append_dbus_tasks(NETWORK, [firewall_dbus_task])
@@ -167,7 +167,7 @@ def _prepare_configuration(payload, ksdata):
     generate_initramfs.append(Task("Generate initramfs", run_generate_initramfs))
     configuration_queue.append(generate_initramfs)
 
-    if is_module_available(SECURITY):
+    if False:
         security_proxy = SECURITY.get_proxy()
 
         # Configure FIPS.
diff --git a/pyanaconda/kickstart.py b/pyanaconda/kickstart.py
index f0ff16145..67297af70 100644
--- a/pyanaconda/kickstart.py
+++ b/pyanaconda/kickstart.py
@@ -357,6 +357,11 @@ class AnacondaKickstartSpecification(KickstartSpecification):
         command_map = dict(cls.commands)
 
         for name, command in handler.commandMap.items():
+            # Skip unsupported commands
+            if name in ('auth', 'authconfig', 'authselect', 'firewall',
+                        'firstboot'):
+                continue
+
             # Ignore removed commands.
             if issubclass(command, RemovedCommand):
                 continue
-- 
2.7.4

